test/integration/smoke/test_network_ipv6.py

```diff
@@ -0,0 +1,357 @@
+        cls.services = testClient.getParsedTestDataConfig()
+        cls._cleanup = []
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cls.apiclient = super(
+                TestCreateIpv6NetworkOffering,
+                cls).getClsTestClient().getApiClient()
+            # Clean up, terminate the created templates
+            cleanup_resources(cls.apiclient, cls._cleanup)
+
+        except Exception as e:
+            raise Exception("Warning: Exception during class cleanup : %s" % e)
+
+    @attr(
+        tags=[
+            "advanced",
+            "basic",
+            "eip",
+            "sg",
+            "advancedns",
+            "smoke"],
+        required_hardware="false")
+    def test_01_create_ipv6_network_offering(self):
+        """Test to create network offering
+
+        # Validate the following:
+        # 1. createNetworkOfferings should return valid info for new offering
+        # 2. The Cloud Database contains the valid information
+        """
+        ipv6_service = self.services["network_offering"]
+        ipv6_service["internetprotocol"] = "dualstack"
+        network_offering = NetworkOffering.create(
+            self.apiclient,
+            ipv6_service
+        )
+        self.cleanup.append(network_offering)
+
+        self.debug("Created Network offering with ID: %s" % network_offering.id)
+
+        cmd = listNetworkOfferings.listNetworkOfferingsCmd()
+        cmd.id = network_offering.id
+        list_network_off_response = self.apiclient.listNetworkOfferings(cmd)
+        self.assertEqual(
+            isinstance(list_network_off_response, list),
+            True,
+            "Check list response returns a valid list"
+        )
+        self.assertNotEqual(
+            len(list_network_off_response),
+            0,
+            "Check Network offering is created"
+        )
+        network_off_response = list_network_off_response[0]
+
+        self.assertEqual(
+            network_off_response.id,
+            network_offering.id,
+            "Check server id in createNetworkOffering"
+        )
+        self.assertEqual(
+            network_off_response.details.internetProtocol.lower(),
+            ipv6_service["internetprotocol"].lower(),
+            "Check internetprotocol in createNetworkOffering"
+        )
+        return
+
+class TestIpv6Network(cloudstackTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        testClient = super(TestIpv6Network, cls).getClsTestClient()
+        cls.services = testClient.getParsedTestDataConfig()
+        cls.apiclient = testClient.getApiClient()
+        cls.dbclient = testClient.getDbConnection()
+        cls._cleanup = []
+
+        cls.zone = get_zone(cls.apiclient, testClient.getZoneForTests())
+        cls.services['mode'] = cls.zone.networktype
+        cls.ipv6NotSupported = False
+
+
+        cmd = listGuestNetworkIpv6Prefixes.listGuestNetworkIpv6PrefixesCmd()
```

Comment:
@shwstppr 
can you add test ipv6 ranges to test_data.py and create ipv6 range in this test ?

Reply:
@weizhouapache sure, will add that